File: backend/CncApp/CncApp.Application/Dtos/SupervisorDashboard/SupervisorDashboardOperatorDto.cs
namespace CncApp.Application.Dtos.SupervisorDashboard;

public class SupervisorDashboardOperatorDto
{
    public int OperatorId { get; set; }
    public string OperatorName { get; set; } = string.Empty;
    public int MachinesRunning { get; set; }
    public List<SupervisorDashboardActiveJobDto> ActiveJobs { get; set; } = new();
    public int PartsMadeToday { get; set; }
    public int ScrapToday { get; set; }
    public decimal ScrapPercentage { get; set; }
}

File: frontend/angular/src/app/core/dtos/supervisor-dashboard/supervisor-dashboard-operator.dto.ts
import { SupervisorDashboardActiveJobDto } from './supervisor-dashboard-active-job.dto';

export interface SupervisorDashboardOperatorDto {
  operatorId: number;
  operatorName: string;
  machinesRunning: number;
  activeJobs: SupervisorDashboardActiveJobDto[];
  partsMadeToday: number;
  scrapToday: number;
  scrapPercentage: number;
}
